packages/browser-wallet/src/shared/utils/verifiable-credential-helpers.ts

```diff
@@ -635,3 +649,70 @@ export async function getChangesToCredentialSchemas(
     }
     return { data: updatedSchemasInStorage, updateReceived };
 }
+
+/**
+ * Get the registry issuer public key from a credential registry CIS-4 contract.
+ * @param client the GRPC client for accessing a node
+ * @param contractAddress the address of a CIS-4 contract
+ * @returns the registry public key for the contract
+ */
+export async function getCredentialRegistryIssuerKey(
+    client: ConcordiumGRPCClient,
+    contractAddress: ContractAddress
+): Promise<string> {
+    const instanceInfo = await client.getInstanceInfo(contractAddress);
+    if (instanceInfo === undefined) {
+        throw new Error('Given contract address was not a created instance');
+    }
+
+    const result = await client.invokeContract({
+        contract: contractAddress,
+        method: `${getContractName(instanceInfo)}.issuer`,
+    });
+
+    if (result.tag !== 'success') {
+        throw new Error(result.reason.tag);
+    }
+
+    const { returnValue } = result;
+    if (returnValue === undefined) {
+        throw new Error(`Return value is missing from issuer public key result in CIS-4 contract: ${contractAddress}`);
+    }
+
+    return returnValue;
+}
+
+/**
+ * Create a publicKey DID identitifer for the given key.
+ */
+export function createPublicKeyIdentifier(publicKey: string, network: NetworkConfiguration): string {
+    return `did:ccd:${getNet(network).toLowerCase()}:pkc:${publicKey}`;
+}
+
+/**
+ * Create a DID identitifer for the given web3Id credential.
+ */
+export function createCredentialId(
+    credentialHolderId: string,
+    issuer: ContractAddress,
+    network: NetworkConfiguration
+): string {
+    return `did:ccd:${getNet(network).toLowerCase()}:sci:${issuer.index}:${
+        issuer.subindex
+    }/credentialEntry/${credentialHolderId}`;
+}
+
+/**
+ * Extracts the credential registry contract addres from a verifiable credential id (did).
+ * @param credentialId the did for a credential
+ * @returns the contract address of the issuing contract of the provided credential id
+ */
+export function getDIDnetwork(did: string): 'mainnet' | 'testnet' {
+    const didParts = did.split(':');
+    const network = didParts[2];
+    if (network !== 'testnet' && network !== 'mainnet') {
+        // Only testnet and mainnet are valid identifiers, and if neither are present, then the network identifier is not present, and it defaults to mainnet
```

Comment:
It would be nice to refer to the documentation that defines that this is the case: https://proposals.concordium.software/ID/concordium-did.html#identifier-syntax.